framework/src/modules/auth/utils.ts

```diff
@@ -175,13 +178,56 @@ export const verifySingleSignatureTransaction = (
 	);
 };
 
+export const verifySignatures = (
+	authModuleId: Buffer,
+	transaction: Transaction,
+	transactionBytes: Buffer,
+	networkIdentifier: Buffer,
+	account: AuthAccount,
+) => {
+	if (
+		transaction.moduleID.equals(authModuleId) &&
+		transaction.commandID.readInt32BE(0) === COMMAND_ID_DELEGATE_REGISTRATION
+	) {
+		verifyRegisterMultiSignatureTransaction(
+			TAG_TRANSACTION,
+			registerMultisignatureParamsSchema,
+			transaction,
+			transactionBytes,
+			networkIdentifier,
+		);
+		return { verified: true };
+	}
+
+	// Verify multi signature registration transaction
+	if (!isMultisignatureAccount(account)) {
+		verifySingleSignatureTransaction(
+			TAG_TRANSACTION,
+			transaction,
+			transactionBytes,
+			networkIdentifier,
+		);
+		return { verified: true };
+	}
+
+	verifyMultiSignatureTransaction(
+		TAG_TRANSACTION,
+		networkIdentifier,
+		transaction.id,
+		account,
+		transaction.signatures,
+		transactionBytes,
+	);
+	return { verified: true };
+};
+
 export const verifyNonce = (
 	transaction: Transaction,
 	senderAccount: AuthAccount,
 ): VerificationResult => {
-	if (transaction.nonce < senderAccount.nonce) {
+	if (transaction.nonce !== senderAccount.nonce) {
```

Comment:
Transaction pool allows transactions to arrive with nonce greater than since transactions don't always distribute in the correct order.  Therefore can we verify that this logic be changed to `!==`

Reply:
Transaction pool allows transactions to arrive with nonce greater than since transactions don't always distribute in the correct order. Therefore can we verify that this logic be changed to !==